src/softmax.jl

```diff
@@ -22,6 +22,20 @@ with additional manipulations enhancing numerical stability.
 For a matrix input `x` it will by default (`dims = 1`) treat it as a batch of vectors,
 with each column independent. Keyword `dims = 2` will instead treat rows independently, and so on.
 
+Since softmax by default acts on a vector instead of a scalar, a syntax different from scalar
```

Comment:
This is great but should say "array" not vector, I think.

Reply:
Maybe it's OK, thinking more. 

```suggestion
Since softmax by default acts on whole columns of array, not individual nubmers, a syntax different that for from ordinary scalar
```